src/modules.rs

```diff
@@ -575,3 +603,17 @@ fn is_cfg_if(item: &ast::Item) -> bool {
         _ => false,
     }
 }
+
+fn is_cfg_match(item: &ast::Item) -> bool {
+    match item.kind {
+        ast::ItemKind::MacCall(ref mac) => {
+            if let Some(last_segment) = mac.path.segments.last() {
```

Comment:
I want to draw attention to this: `cfg_match!` is checking that the *last* path segment is `cfg_match!` (i.e. tries to process `any::path::cfg_match!`), whereas `cfg_if!` is checking that the *first* path segment is `cfg_if!` (i.e. tries to process `cfg_if::any::path!`). I *think* my behavior is more correct, and the worst case scenario is that some more files get discovered when a custom `cfg_match!` that looks similar enough to `core::cfg_match!` is used, but it bears pointing out.

If factoring this behavior out to be the same between the two is desirable, I can do that, and would choose to check the macro-ident, not the path front. (The current `cfg_if!` processing will skip `::cfg_if::cfg_if!`, since the path starts with `kw::PathRoot`.)

Reply:
I went ahead and wrote the patch to unify the predicate: a129548a6a7d7a040f4aa5607e77ae5126226dc3

I haven't included it in this PR since it's behavior changing for the existing `cfg_if!` support, to keep this PR scoped to just the `cfg_match!` support.

Reply:
... and I ended up doing that cleanup as well: 4e40b2c5fb4c63a7a18b0fc9d07e983c643734fc

But I'm actually done doing extended cleanup for this now 😅

To be fully explicit about it: the two linked commits in this and the prior comment are made available under the same license terms as if I had directly contributed them by placing them in a PR. Feel free to pull them (or ask me to PR them) if you believe them to be an improvement.